cladecombiner/aggregator.py

```diff
@@ -0,0 +1,278 @@
+from abc import ABC, abstractmethod
+from collections.abc import Iterable
+from copy import copy
+from dataclasses import dataclass
+from warnings import warn
+
+from .ordered_taxon import OrderedTaxon
+from .taxon import Taxon
+from .taxonomy_scheme import PhylogeneticTaxonomyScheme
+from .utils import table
+
+
+@dataclass
+class TaxonMapping:
+    """
+    Class for tracking proposed step in aggregation process.
+    """
+
+    map: dict[Taxon, Taxon]
+    "from : to dictionary of the mapping"
+    done: dict[Taxon, bool]
+    "to : add_to_stack dictionary tracking if the aggregated taxon should be added back to the stack"
```

Comment:
like, `done = False` means keep in the "stack" of taxa to be aggregated?

Reply:
Yep